r/tests/testthat/test-dplyr-mutate.R

```diff
@@ -415,3 +406,160 @@ test_that("mutate and write_dataset", {
       summarize(mean = mean(integer))
   )
 })
+
+# PACHA ADDITIONS ----
+# READ THIS CAREFULLY PLEASE, IT'S MY 1ST DAY WRITING THIS KIND OF SENSITIVE TESTS
+
+# similar to https://github.com/tidyverse/dplyr/blob/master/tests/testthat/test-mutate.r#L1-L10
+# the rest of that test belongs in L55-62 here
+test_that("empty mutate returns input", {
+  # dbl2 = 5, so I'm grouping by a constant
+  gtbl <- group_by(tbl, dbl2)
+
+  expect_dplyr_equal(input %>% mutate(!!!list()) %>% collect(), tbl)
+
+  expect_dplyr_equal(input %>% mutate() %>% collect(), gtbl)
+  expect_dplyr_equal(input %>% mutate(!!!list()) %>% collect(), gtbl)
+})
+
+# similar to https://github.com/tidyverse/dplyr/blob/master/tests/testthat/test-mutate.r#L12-L6
+test_that("rownames preserved", {
+  skip("Row names are not preserved")
+  df <- data.frame(x = c(1, 2), row.names = c("a", "b"))
+  expect_dplyr_equal(input %>% mutate(y = 2) %>% collect(), df)
+})
+
+# similar to https://github.com/tidyverse/dplyr/blob/master/tests/testthat/test-mutate.r#L18-L29
+test_that("mutations applied progressively", {
+  df <- tibble(x = 1)
+
+  expect_dplyr_equal(
+    input %>% mutate(y = x + 1, z = y + 1) %>% collect(),
+    df
+  )
+  expect_dplyr_equal(
+    input %>% mutate(x = x + 1, x = x + 1) %>% collect(),
+    df
+  )
+  expect_dplyr_equal(
+    input %>% mutate(y = x + 1, z = y + 1) %>% collect(),
+    df
+  )
+
+  df <- data.frame(x = 1, y = 2)
+  expect_dplyr_equal(
+    input %>% mutate(x2 = x, x3 = x2 + 1) %>% collect(),
+    df
+  )
```

Comment:
For this one, you can't use `expect_dplyr_equal()` if you want to test what the test at https://github.com/tidyverse/dplyr/blob/master/tests/testthat/test-mutate.r#L24-L28 tests. Better to just use `expect_equal()` for this particular one:
```suggestion
  df <- data.frame(x = 1, y = 2)
  expect_equal(
    df %>% Table$create() %>% mutate(x2 = x, x3 = x2 + 1) %>% collect(),
    df %>% Table$create() %>% mutate(x2 = x + 0, x3 = x2 + 1) %>% collect()
  )
```